v2/example_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019, Google Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package gax
+
+import (
+	"time"
+
+	"google.golang.org/grpc/codes"
+)
+
+// Some result that the client might return.
+type fakeResponse struct {}
+
+// Some client that can perform RPCS.
+type fakeClient struct {}
+
+// Some fake RPC that a client might perform.
+func (c *fakeClient) PerformSomeRPC() (*fakeResponse, error) {
+	// An actual client would return something meaningful here.
+	return nil, nil
+}
+
+func ExampleOnCodes() {
+	c := &fakeClient{}
+
+	// INTERNAL and UNKNOWN are typically safe to retry for idempotent RPCs.
+	retryer := OnCodes([]codes.Code{codes.Unknown, codes.Unavailable}, Backoff{
+		Initial: time.Second,
+		Max: 32 * time.Second,
+		Multiplier: 2,
+	})
+
+	var resp *fakeResponse
+	for {
+		var err error
+		resp, err = c.PerformSomeRPC()
+		if err != nil {
+			if delay, shouldRetry := retryer.Retry(err); shouldRetry {
+				time.Sleep(delay)
+				continue
+			}
+			// TODO: handle error
+		}
+		break
+	}
+	_ = resp // TODO: use resp
```

Comment:
It's actually not a sure thing that we can use `resp` here, since we might have failed un-retry-ably.

I think maybe this?
```
if err == nil {
  break // use resp
}
delay, shouldRetry := retryer.Retry(err)
if !shouldRetry {
  break // use err
}
gax.Sleep(ctx, delay)
```

Reply:
1. There's some subtleties about declaring global errs that I was trying to avoid. I went ahead and stuck the retry block in its own function to make it a lot cleaner and clearer, at the expense of more verbose code. WDYT? 
2. Trying to avoid gax.Sleep so that users only have one thing they're learning about per example.

Reply:
Done